### ISSUE TITLE
Update contentful/starter-gatsby-blog to use Gatsby  2

### ISSUE THREAD

**Issue by chrfritsch** [contentful/starter-gatsby-blog#8]
Gatsby 2 was released a month ago. So it would be nice if this starter would use it.

https://www.gatsbyjs.org/blog/2018-09-17-gatsby-v2/

**Comment by bytheway875** [comment]
i took a stab at upgrading to v2 gatsby and got it working with a couple of caveats:

- The .babelrc - I wasn't sure what it was being used for (In most of my gatsby apps, the babelrc is only for tests and this app didn't have tests), and I was getting the error ` TypeError: htmlComponentRenderer.default is not a function` until I deleted the .babelrc. Have not seen any issues after deleting.

- The graphql query for author didn't work. For whatever reason, 
```
    allContentfulPerson(filter: { id: { eq: "validIdThatWorkedinV1" } }) {
```
did not work. However, changing it to 
```
    allContentfulPerson(filter: { name: { eq: "Author Name" } }) {
```

worked.

I think that the problem with that lies somewhere in this update: 

https://github.com/gatsbyjs/gatsby/pull/3807/files

It's noted in the upgrade path that "Source and transformer plugins now use UUIDs for IDs. If you used glob or regex to query nodes by id then you'll need to query something else." I'm not quite sure what that means,  but I think whats happening is that the contentful gatsby plugin is now reassigning the id node to a uuid created in the plugin, so we can't query by id(?). Trying to get more clarification on that...

Otherwise, I transitioned the layout/index.js file into a Layout component in the components directory since layouts are no longer supported, and am also going to try an implementation with the gatsby-plugin-layout to prevent navigation re-rendering as suggested in the docs.

Otherwise, I was able to develop, build, and deploy as usual after the updates! Will create a PR.

**Comment by bytheway875** [comment]
Hmm, I actually made my updates off of https://github.com/contentful-userland/gatsby-contentful-starter which diverges a little bit more from this repo than I thought.

**Comment by stefanjudis** [comment]
@bytheway875 Yeah this one is version that runs in our official in-app onboarding. Userland is a community maintained one where community members also get commit access. :)

Btw there is an open v2 PR already. :)

https://github.com/contentful-userland/gatsby-contentful-starter/pull/34

**Comment by arpl** [comment]
Fixed in #47
